### PR TITLE
Binary Serialization Standard

### LINKED ISSUE #1471: Update Serialization Standard
Update the Serialization Standard's format to clarify larger internal structures and add new types for Condor.

Change title to Binary Serialization Standard

### PR DESCRIPTION
**Pull request by ACStoneCL**
### What does this PR fix/introduce?
This PR turns the "Serialization Standard" into the "Binary Serialization Standard" and splits it into multiple pages for ease of reading.

The document itself now includes all updated `Condor` types outside of the chainspec, which will be a separate PR.

Closes #1471 #1378 

### Additional context
// Add further detail on the current situation, or what the PR seeks to address.
// Include [loom](https://www.loom.com/), screenshots, or gifs([record](https://giphy.com/apps/giphycapture), [compress](https://gifcompressor.com/)) if helpful.

### Checklist
(Delete any that aren't relevant)

- [x] Docs are successfully building - `yarn install && yarn run build`.
- [x] For new **internal** links I used *relative file paths* (with .md extension) - e.g. `../../faq/faq-general.md` - instead of introducing *absolute file path*, or *relative/absolute URL*.
- [x] All external links have been verified with `yarn run check:externals`.
- [x] My changes follow the [Casper docs style guidelines](https://docs.casper.network/resources/contribute-to-docs/).
- [x] All technical procedures have been tested (if you want help with this, mention it in [Reviewers](#reviewers)).
- [x] If structural changes are introduced (not just content changes), cross-broswer testing has been completed.

### Reviewers
TBD

### CODE REVIEW

**Comment by ACStoneCL** [comment]
@bradjohnl I'm getting a strange failure on the CICD for this one - could you take a look when you get a chance?

**Comment by bradjohnl** [comment]
> @bradjohnl I'm getting a strange failure on the CICD for this one - could you take a look when you get a chance?

Hey Adam. It's fixed. I've removed some hardcoded, and apparently redundant tests that were targeting pages that this PR removed / changed.

I've created an internal issue in our SRE board to check the testing strategy and make sure we are already testing each internal and external link with other processes (namely the: node check-external-urls.js test script).

**Review by ipopescu — COMMENTED** [review]
I left a few minor nits and a few questions about some methods I don't see anymore. It would be good to address these before approving.

While reviewing this file, a few other questions came up:
- There's a bit of duplication with the json-rpc/types_chain.md file, although the serialization section is much more detailed. Perhaps we can auto-generate the json-rpc docs in the future to avoid double maintenance.
-  I see that some types are missing from what we expose in the JSON RPC; perhaps this is because they are not exposed in the RPC methods. Do you know if this the case?
	- Some of the fields I noted were: Message, MessageAddr, MessageLimits, MessagePayload, MessageTopicOperation, DeployConfig, TransactionConfig, TransactionHeader, TransactionId, TransactionWithExecutionInfo, BalanceHoldAddr, BidAddr, BlockGlobalAddr, BlockSignatures.

**Comment by ipopescu** [comment]
I tried to help resolve a conflict, and then I had to fix a link for the build to pass.

File: source/docs/casper/concepts/serialization/types.md [hunk 1]
```diff
@@ -0,0 +1,910 @@
...
+
+-   `effect`: The [effect](#executioneffect) of executing the deploy.
+
+-   `transfers`: A record of transfers performed while executing the deploy, serialized as a [`List`](./primitives.md#clvalue-list).
+
+-   `cost`: The cost of executing the deploy, serializes as a [`U512`](./primitives.md#clvalue-numeric) value.
+
+## ExecutionResultV2 {#executionresultv2}
+
+The result of a single deploy. It serializes as a `u8` tag indicating either `Failure` as a 0 or `Success` as a 1. This is followed by the appropriate structure below:
```
Comment (ipopescu) [inline source/docs/casper/concepts/serialization/types.md:445]: I think this should refer to transactions instead of deploys.

**Review by darthsiroftardis — APPROVED** [review]
Approval pending comments from @ipopescu

File: source/docs/casper/concepts/serialization/structures.md [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+    -   `0x03b348fdd0d0b3f66468687df93141b5924f6bb957d5893c08b60d5a78d0b9a423001400000050734c7a3563374a73715438424b386c6c306b463b0000003d0d7f193f70740386cb78b383e2e30c4f976cf3fa834bafbda4ed9dbfeb52ce1777817e8ed8868cfac6462b7cd31028aa5a7a60066db35371a2f8`
+
+-   StoredVersionedContractByName serializes such that the first byte within the serialized buffer is 4u8. The name and entry_point are serialized as a [String](./primitives.md#string-clvalue-string) CLValue, with the version field serializing as an [Option](./primitives.md#option-clvalue-option).
+
+    -   `StoredVersionedContractByName { name: "lWJWKdZUEudSakJzw1tn", version: Some(1632552656), entry_point: "S1cXRT3E1jyFlWBAIVQ8", args: 9975e6957ea6b07176c7d8471478fb28df9f02a61689ef58234b1a3cffaebf9f303e3ef60ae0d8 }`
+    -   `0x04140000006c574a574b645a5545756453616b4a7a7731746e01d0c64e61140000005331635852543345316a79466c57424149565138270000009975e6957ea6b07176c7d8471478fb28df9f02a61689ef58234b1a3cffaebf9f303e3ef60ae0d8`
```
[the remark this answers is not in the document]
Comment (ACStoneCL) [reply source/docs/casper/concepts/serialization/structures.md:292]: I'm going to remove this section entirely, as the new `docs-redux` repo will have it available as part of the versioning.

File: source/docs/casper/concepts/serialization/structures.md [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+The body of a `TransactionV1`, consisting of the following:
+
+- [`args`](./types.md#runtimeargs-runtimeargs)
+- [`target`](#transactiontarget)
+- [`entry_point`](#transactionentrypoint)
+- [`scheduling`](#transactionscheduling)
+
+### TransactionRuntime {#transactionruntime}
+
+The runtime used to execute a transaction, serialized as a [`u8`](./primitives.md#numeric-clvalue-numeric). Currently, only the `VmCasperV1` is available, which serializes as a `0`.
```
[the remark this answers is not in the document]
Comment (ACStoneCL) [reply source/docs/casper/concepts/serialization/structures.md:367]: I think that would be something we can update at a later time. I don't think it's in any state for inclusion as is.

File: source/docs/casper/concepts/serialization/types.md [hunk 4]
```diff
@@ -0,0 +1,910 @@
...
+
+-   `era_id`: An [Era ID newtype](#eraid) identifying the era when the protocol version will activate.
+
+-   `timestamp`: A [timestamp](#timestamp) if the activation point is of the Genesis variant.
+
+## AddressableEntityHash {#addressable-entity-hash}
+
+The hex-encoded address of an addressable entity, which serializes as the byte representation of itself.
+
+## Approval {#approval}
```
[the remark this answers is not in the document]
Comment (ACStoneCL) [reply source/docs/casper/concepts/serialization/types.md:47]: I've moved it over into structures.md fully now.

File: source/docs/casper/concepts/serialization/types.md [hunk 5]
```diff
@@ -0,0 +1,910 @@
...
+
+-   `ModuleBytes`: Serializes as a `u8` tag of 0 followed by [`bytes`](#bytes) and [`runtimeargs`](#runtimeargs).
+
+-   `StoredContractByHash`: Serializes as a `u8` tag of 1 followed by the contract hash as an [`AddressableEntityHash`](./structures.md#addressable-entity-hash), the name of an entry point as a [String](./primitives.md#string-clvalue-string) and [`runtimeargs`](#runtimeargs).
+
+-   `StoredContractByName`: Serializes as a `u8` tag of 2 followed by the named key as a [String](./primitives.md#string-clvalue-string), the name of an entry point as a `String` and [`runtimeargs`](#runtimeargs).
+
+-   `StoredVersionedContractByHash`: Serializes as a `u8` tag of 3 followed by the [`PackageHash`](#package-hash), the `version` as an [`Option`](./primitives.md#clvalue-option), an entry point as a [String](./primitives.md#string-clvalue-string) and [`runtimeargs`](#runtimeargs).
+
+-   `StoredVersionedContractByName`: Serializes as a `u8` tag of 4 followed by the named key as a [String](./primitives.md#string-clvalue-string), the `version` as an [`Option`](./primitives.md#clvalue-option), the name of an entry point as a `String` and [`runtimeargs`](#runtimeargs).
```
[the remark this answers is not in the document]
Comment (ACStoneCL) [reply source/docs/casper/concepts/serialization/types.md:405]: I'm still seeing them in the code, so I'll leave this part for now.

File: source/docs/casper/concepts/serialization/types.md (hunk 1 above)
Comment (ACStoneCL) [reply source/docs/casper/concepts/serialization/types.md:445]: Good catch!

**Pull request merged**